src/ui/components/atoms/dropdown-stats/dropdown-stats.tsx

```diff
@@ -0,0 +1,147 @@
+import React, { useEffect, useRef } from 'react';
+import './dropdown-stats.scss';
+
+export interface DropdownStatsOption {
+  label: string;
+  value: string;
+  code: string;
+  stat: number;
+}
+
+export interface DropdownStatsProps {
+  options: DropdownStatsOption[];
+  selectedValue?: string;
+  onChange: (value: string) => void;
+  maxValue?: number;
+  barColor?: string;
+  showPercentage?: boolean;
+  className?: string;
+  disabled?: boolean;
+  isOpen: boolean;
+  onToggle: () => void;
+  displayFormat?: 'percentage' | 'value' | 'both';
+}
+
+function DropdownStats({
+  options,
+  selectedValue,
+  onChange,
+  maxValue,
+  barColor = '#8a2be2',
+  showPercentage = true,
+  className = '',
+  disabled = false,
+  isOpen,
+  onToggle,
+  displayFormat = 'both',
+}: DropdownStatsProps) {
+  const dropdownRef = useRef<HTMLDivElement>(null);
+  const selectedOption = options.find((option) => option.value === selectedValue) ?? options[0];
+  const calculatedMaxValue = maxValue ?? Math.max(...options.map((option) => option.stat), 1);
+
+  useEffect(
+    function setupClickOutsideListener() {
+      function handleClickOutside(event: MouseEvent) {
+        if (dropdownRef.current && !dropdownRef.current.contains(event.target as Node)) {
+          onToggle();
+        }
+      }
+
+      if (isOpen) {
+        document.addEventListener('mousedown', handleClickOutside);
+      }
+
+      return function cleanupClickOutsideListener() {
+        document.removeEventListener('mousedown', handleClickOutside);
+      };
+    },
+    [isOpen, onToggle],
+  );
```

Comment:
This component has many logic Don't be an atom, please move them to a molecule folder